### PR TITLE
fix: table component should be scoped when searching items

### LINKED ISSUE #10210: table component is not working with multiple instances
### Bug description

implementing the task manager, Axel found while testing a layout issue

investigating, I think the flaw is related to how the afterUpdate method is implemented

https://github.com/podman-desktop/podman-desktop/blob/main/packages/ui/src/lib/table/Table.svelte#L148-151

it will update all the matching css parts of the document

### Operating system

macOS

### Installation Method

None

### Version

1.14.1

### Steps to reproduce

you might look at https://github.com/podman-desktop/podman-desktop/pull/10206

### Relevant log output

```shell

```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by benoitf**
### What does this PR do?
Table componet was searching and updating items on the whole document/page
It should be scoped and only search/update in its children

### What issues does this PR fix or reference?

fixes https://github.com/podman-desktop/podman-desktop/issues/10210

### How to test this PR?

you might try this PR on top of the task manager PR but I also added a unit test

- [x] Tests are covering the bug fix or the new feature

### CODE REVIEW

File: packages/ui/src/lib/table/Table.spec.ts [hunk 1]
```diff
@@ -380,3 +380,20 @@ test('Expect duration cell to be empty when undefined', async () => {
     expect(cells[6].textContent?.trim()).toBe(expected[i - 1]);
   }
 });
+
+test('Expect table is scoped for css manipulation', async () => {
```
Comment (dgolovin) [inline packages/ui/src/lib/table/Table.spec.ts:384]: nit: the reason for this change is to keep fix table rendering in the case when multiple tables are visible. The test proves table rendering does not conflict with other UI elements. There should be tests for two tables rendered in the same page.

File: packages/ui/src/lib/table/Table.spec.ts [hunk 2]
```diff
@@ -380,3 +380,20 @@ test('Expect duration cell to be empty when undefined', async () => {
...
+  render(TestTable, {});
+
+  // Wait for the table to update
+  await tick();
+
+  const table = await screen.findByRole('table');
+  expect(table).toBeDefined();
```
Comment (dgolovin) [inline packages/ui/src/lib/table/Table.spec.ts:391]: nit: to be sure that changes does not affect table rendering there should be check for `table.style.gridTemplateColumns` not an empty string.

**Review by dgolovin — APPROVED** [review]
LGTM with nits.

File: packages/ui/src/lib/table/Table.spec.ts (hunk 1 above)
Comment (axel7083) [reply packages/ui/src/lib/table/Table.spec.ts:384]: I share the same feedback, having a MultipleTableTest.svelte component or something similar

File: packages/ui/src/lib/table/Table.spec.ts (hunk 2 above)
Comment (benoitf) [reply packages/ui/src/lib/table/Table.spec.ts:391]: test updated

File: packages/ui/src/lib/table/Table.spec.ts (hunk 1 above)
Comment (benoitf) [reply packages/ui/src/lib/table/Table.spec.ts:384]: test updated

**Comment by benoitf** [comment]
pr updated

**Review by axel7083 — APPROVED** [review]
Thanks for the change and the fix

**Review by dgolovin — APPROVED** [review]
LGTM!
Thank you for update.

**Pull request merged**